app/domain/authentication/authn_k8s/consts.rb

```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+module Authentication
+  module AuthnK8s
+
+    K8S_AUTHENTICATOR_NAME = 'authn-k8s'
+
+    module Restrictions
+
+      NAMESPACE = "namespace"
+      SERVICE_ACCOUNT = "service-account"
+      POD = "pod"
+      DEPLOYMENT = "deployment"
+      STATEFUL_SET = "stateful-set"
+      DEPLOYMENT_CONFIG = "deployment-config"
```

Comment:
A subtle point I noticed when looking at the azure `consts.rb` file too.  What I'm about to say probably falls under "existing code" and is outside the scope of this PR, but is worth mentioning.

This set of variables from line 10-15 do not actually represent restrictions.  As I understand it, these are the string values that will be used as the annotation keys stored in the DB for each of these concepts.  Interestingly, I was not able to find any code that _stored_ annotations in the db referring to them, which means that those references are probably hard-coded somewhere, minimizing the value of this abstraction (keeping the "annotation key strings" in sync between the code that writes the annotations and the code that reads them would be one of the primary benefits of this abstraction).

Reply:
They are hard-coded in the policy of the integration tests.
I agree this is less readable, but this is still an improvement to the existing code.
I'm not going to go any further than this.